File: backend/src/Workora.Application/Features/Documents/DTOs/DocumentDto.cs
using Workora.Domain.Enums;
using Workora.Shared.Responses;

namespace Workora.Application.Features.Documents.DTOs;

/// <summary>
/// DTO representing an uploaded document record.
/// </summary>
public record DocumentDto(
    int Id,
    Guid Uuid,
    int CompanyId,
    int? EmployeeId,
    string? EmployeeName,
    string Title,
    string FileName,
    string FilePath,
    string ContentType,
    long FileSizeBytes,
    DocumentCategory Category,
    bool IsActive,
    DateTimeOffset CreatedAt);

File: frontend/workora-ui/src/app/data/dtos/document.dto.ts
export interface DocumentDto {
  id: number;
  uuid: string;
  companyId: number;
  employeeId?: number | null;
  employeeName?: string | null;
  title: string;
  fileName: string;
  filePath: string;
  contentType: string;
  fileSizeBytes: number;
  category: string;
  isActive: boolean;
  createdAt: string;
}

export interface CreateDocumentRequestDto {
  companyId: number;
  employeeId?: number | null;
  title: string;
  fileName: string;
  filePath: string;
  contentType: string;
  fileSizeBytes: number;
  category: string;
}
